File: doc-examples/nitrite-example-java/src/test/java/example/TransactionServiceTest.java
package example;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
class TransactionServiceTest {

    @Inject
    TransactionService transactionService;

    @Inject
    BookRepository bookRepository;

    @AfterEach
    void cleanup() {
        bookRepository.deleteAll();
    }

    @Test
    void testTransactionalSave() {
        // Save a book within a transaction
        transactionService.saveBook("Transactional Book");

        // Verify the book was saved
        var books = bookRepository.findAll();
        assertEquals(1, books.size());
        assertEquals("Transactional Book", books.get(0).getTitle());
    }

    @Test
    void testNonTransactionalOperation() {
        // Save without transaction
        transactionService.logWithoutTransaction("Non-transactional Book");

        // Verify the book was saved
        var books = bookRepository.findAll();
        assertEquals(1, books.size());
        assertEquals("Non-transactional Book", books.get(0).getTitle());
    }
}

File: doc-examples/nitrite-example-kotlin/src/test/kotlin/example/TransactionServiceTest.kt
package example

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull

@MicronautTest
class TransactionServiceTest {

    @Inject
    lateinit var transactionService: TransactionService

    @Inject
    lateinit var bookRepository: BookRepository

    @AfterEach
    fun cleanup() {
        bookRepository.deleteAll()
    }

    @Test
    fun testTransactionalSave() {
        // Save a book within a transaction
        transactionService.saveBook("Transactional Book")

        // Verify the book was saved
        val books = bookRepository.findAll()
        assertEquals(1, books.size)
        assertEquals("Transactional Book", books[0].title)
    }

    @Test
    fun testNonTransactionalOperation() {
        // Save without transaction
        transactionService.logWithoutTransaction("Non-transactional Book")

        // Verify the book was saved
        val books = bookRepository.findAll()
        assertEquals(1, books.size)
        assertEquals("Non-transactional Book", books[0].title)
    }
}
